### PR TITLE
Add sort() entry to Java ArrayList

### PR DESCRIPTION
**Pull request by Brag2gr8**
### Description

- Added the Calendar .sort() term for Java
- Closing issue #4102 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/java/concepts/array-list/terms/sort/sort.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
+---
+Title: '.sort()'
+Description: 'Used to sort arrays of primitive types and objects'
```
Comment (dakshdeepHERE) [inline content/java/concepts/array-list/terms/sort/sort.md:3]:
```suggestion
Description: 'Used to sort arrays of primitive types and objects.'
```

File: content/java/concepts/array-list/terms/sort/sort.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+## Syntax
+
+```pseudo
+Arrays.sort(array);
+```
+
+Here's a basic example of using Arrays.sort() with an array of integers:
+
+## Example 1
```
Comment (dakshdeepHERE) [inline content/java/concepts/array-list/terms/sort/sort.md:25]:
```suggestion
## Examples

### Example 1
```

File: content/java/concepts/array-list/terms/sort/sort.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  - 'Data Types'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+The **`.sort()`** method is used to sort arrays of primitive types and objects. The sorting is done in ascending order by default, but it can be customized for objects by implementing the `Comparable` interface or using a `Comparator`.
+
+## Syntax
+
```
Comment (dakshdeepHERE) [inline content/java/concepts/array-list/terms/sort/sort.md:18]:
```suggestion

Here's a basic syntax of how to use the `.sort()` method:

```

File: content/java/concepts/array-list/terms/sort/sort.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+```pseudo
+Arrays.sort(array);
+```
+
+Here's a basic example of using Arrays.sort() with an array of integers:
+
+## Example 1
+
+In this case, strings implement the `Comparable` interface, so they can be sorted directly using `Arrays.sort()`.
```
Comment (dakshdeepHERE) [inline content/java/concepts/array-list/terms/sort/sort.md:27]:
```suggestion
In this case, strings implement the `Comparable` interface, so they can be sorted directly using `Arrays.sort()`:
```

File: content/java/concepts/array-list/terms/sort/sort.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    public static void main(String[] args) {
+        int[] numbers = {5, 2, 8, 1, 3};
+
+        // Sorting the array in ascending order
+        Arrays.sort(numbers);
+
+        // Printing the sorted array
+        System.out.println("Sorted Array: " + Arrays.toString(numbers));
+    }
+}
```
Comment (dakshdeepHERE) [inline content/java/concepts/array-list/terms/sort/sort.md:42]:
```suggestion
import java.util.Arrays;

public class Main {
  public static void main(String[] args) {
    int[] numbers = {5, 2, 8, 1, 3};

    // Sorting the array in ascending order
    Arrays.sort(numbers);
    
    System.out.println("Sorted Array: " + Arrays.toString(numbers));
  }
}
```

File: content/java/concepts/array-list/terms/sort/sort.md [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+```
+
+The output should look like this:
```
Comment (dakshdeepHERE) [inline content/java/concepts/array-list/terms/sort/sort.md:45]:
```suggestion
The above example will give the following output:
```

File: content/java/concepts/array-list/terms/sort/sort.md [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+
+```shell
+Sorted Array: [1, 2, 3, 5, 8]
+```
+
+> **Note:** If you need a custom sorting order, you can use the Comparator interface. Here's an example sorting strings by their length:
+
+## Example 2
+
+
```
Comment (dakshdeepHERE) [inline content/java/concepts/array-list/terms/sort/sort.md:55]: have a small introductory sentence here too like you did above

File: content/java/concepts/array-list/terms/sort/sort.md [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+    public static void main(String[] args) {
+        String[] names = {"Alice", "Bob", "Charlie", "David"};
+
+        // Sorting the array of strings in ascending order
+        Arrays.sort(names);
+
+        // Printing the sorted array
+        System.out.println("Sorted Array by Length: " + Arrays.toString(names));
+    }
+}
```
Comment (dakshdeepHERE) [inline content/java/concepts/array-list/terms/sort/sort.md:69]:
```suggestion
import java.util.Arrays;

public class Main {
  public static void main(String[] args) {
    String[] names = {"Alice", "Bob", "Charlie", "David"};

    // Sorting the array of strings in ascending order
    Arrays.sort(names);
    
    System.out.println("Sorted Array by Length: " + Arrays.toString(names));
  }
}
```

File: content/java/concepts/array-list/terms/sort/sort.md [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+```
+
+The output should look like this:
```
Comment (dakshdeepHERE) [inline content/java/concepts/array-list/terms/sort/sort.md:72]:
```suggestion
The above example will give the following output:
```

File: content/java/concepts/array-list/terms/sort/sort.md [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+
+```shell
+Sorted Array: [1, 2, 3, 5, 8]
+```
+
+> **Note:** If you need a custom sorting order, you can use the Comparator interface. Here's an example sorting strings by their length:
```
Comment (dakshdeepHERE) [inline content/java/concepts/array-list/terms/sort/sort.md:51]: We dont allow the use third person and pronouns in our docs contributions. Please refactor this sentence. Dont use words like He she you us ours etc in entries please.

**Review by dakshdeepHERE — CHANGES_REQUESTED** [review]
I have some comments for you.

**Comment by Brag2gr8** [comment]
@dakshdeepHERE I've added the changes. please check it out

**Review by dakshdeepHERE — APPROVED** [review]
@Brag2gr8 I did some minor changes from my side. This looks good for second review. 🚀

**Pull request closed without merging**